src/resolve/resolved-mdns.c

```diff
@@ -386,12 +386,6 @@ static int on_mdns_packet(sd_event_source *s, int fd, uint32_t revents, void *us
         if (r <= 0)
                 return r;
 
-        /* Refuse traffic from the local host, to avoid query loops. However, allow legacy mDNS
-         * unicast queries through anyway (we never send those ourselves, hence no risk).
-         * i.e. check for the source port nr. */
-        if (p->sender_port == MDNS_PORT && manager_packet_from_local_address(m, p))
-                return 0;
```

Comment:
Dunno. I think https://github.com/systemd/systemd/pull/32991 should be reverted and then it would be great to figure out what `resolved` should do in general with queries coming from the same machine. As far as I understand this PR fixes that side effect with replies coming from the same machine but since it still allows queries scenarios where several mDNS responders on the same machine are run are still messed up in the sense that conflicts can't be resolved properly and so on. For example I can see things like
```
Host name conflict, retrying with fedora-10
```
and so on. Without this patch it doesn't happen.

Reply:
The message come from avahi, right? What is the setup? Both Avahi and resolved running on the same system?? Is it supported? Please provide more details.

Reply:
I think it came from avahi but it probably can be any other embedded mDNS responder (it isn't uncommon to embed the mDNS stack into bespoke applications announcing their stuff to other devices).

> Is it supported?

This sort of thing is discouraged because it for example messes up direct unicast queries but people running those things don't care about that usually. I think resolved doesn't respond to direct unicast queries properly according to the RFC anyway.

Either way I don't use resolved  anywhere so I can live with anything. I noticed that regression because I didn't fully turn off the resolved fuzzer at the time and resolved started crashing on the host (and in the container) even though it should have crashed in the container only. I no longer fuzz resolved so it isn't an issue to me either.

Reply:
OK. I will drop the change to accept queries from local host, at least tentatively.

I just read the RFC only briefly, not in detail, but I could not find anything that we should refuse or accept queries from local host. Do you know anything about that? If there is nothing, I think it is OK to refuse them for safety.

Reply:
> Do you know anything about that?

I think it depends on what resolved is supposed to accomplish. As far as I can remember some IPP-over-USB stuff expects things to be advertised on 127.0.0.1/::1 but it wouldn't work even if resolved replied to local queries (because it doesn't even listen on those addresses). I think in its current form it just shouldn't try to reply to anything locally.